File: server/desktop/src/main/java/dev/slimevr/desktop/platform/linux/SocketUtils.java
package dev.slimevr.desktop.platform.linux;

import java.io.IOException;
import java.net.StandardProtocolFamily;
import java.net.UnixDomainSocketAddress;
import java.nio.channels.SocketChannel;


public class SocketUtils {

	static boolean isSocketInUse(String socketPath) {
		try (SocketChannel testChannel = SocketChannel.open(StandardProtocolFamily.UNIX)) {
			testChannel.connect(UnixDomainSocketAddress.of(socketPath));
			return true;
		} catch (IOException e) {
			return false;
		}
	}

}

File: server/desktop/src/main/java/dev/slimevr/desktop/platform/linux/SocketUtils.kt
package dev.slimevr.desktop.platform.linux

import java.io.IOException
import java.net.StandardProtocolFamily
import java.net.UnixDomainSocketAddress
import java.nio.channels.SocketChannel

object SocketUtils {
	fun isSocketInUse(socketPath: String) = try {
		SocketChannel.open(StandardProtocolFamily.UNIX).use { testChannel ->
			testChannel.connect(UnixDomainSocketAddress.of(socketPath))
			true
		}
	} catch (_: IOException) {
		false
	}
}
